src/meshes/t8code_mesh.jl

```diff
@@ -807,42 +824,26 @@ function adapt_callback_wrapper(forest,
 end
 
 """
-    Trixi.adapt!(mesh::T8codeMesh, adapt_callback; kwargs...)
+    Trixi.adapt_forest(forest::Ptr{t8_forest}, adapt_callback; kwargs...)
```

Comment:
Why did the name of this function change? It should have a `!` at the end to signal that its mutating its argument.

Reply:
This change highlighting is a bit misleading. It is actually part of a completely new doc string. I decided to streamline naming conventions:
```
adapt_forest -> accepts old forest, returns new forest
adapt! -> accepts a mesh object, updates it with the new forest

balance_forest -> accepts old forest, returns new forest
balance! -> accepts a mesh object, updates it with the new forest

partition_forest -> accepts old forest, returns new forest
partition! -> accepts a mesh object, updates it with the new forest
```

Reply:
I see, thanks 👍